src/core/packages/saved-objects/migration-server-internal/src/run_v2_migration.ts

```diff
@@ -96,13 +138,14 @@ export const runV2Migration = async (options: RunV2MigrationOpts): Promise<Migra
 
   // compare indexTypesMap with the one present (or not) in the .kibana index meta
   // and check if some SO types have been moved to different indices
-  const indicesWithRelocatingTypes = await getIndicesInvolvedInRelocation({
-    mainIndex: options.kibanaIndexPrefix,
-    client: options.elasticsearchClient,
-    indexTypesMap,
-    logger: options.logger,
-    defaultIndexTypesMap: options.defaultIndexTypesMap,
-  });
+  const indicesWithRelocatingTypes = indexDetails
+    ? // the .kibana index exists, we might have to relocate some SO to different indices
+      getIndicesInvolvedInRelocation(
```

Comment:
The `getIndicesInvolvedInRelocation` method has been simplified and it no longer is `async`.

I use the opportunity of the newly introduced _version check_ to fetch index meta information as well, which is then used by the method to determine if some SO types need to be relocated to a separate index.

NB that this is mostly dead code as we no longer allow reindexing migrations, and it can probably be removed in the scope of the unification of V2 and ZDT.